interview-questions.md

```diff
@@ -7,81 +7,83 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 5. What is React?
 
-  Your answer:
+  Your answer: React is a JavaScript library that is used to create highly responsive and dynamic user interfaces.
 
-  Researched answer:
+  Researched answer: Should have added that it's component based. 
 
 
 
 6. What is yarn? What file(s) are updated when you run `$ yarn`?
 
-  Your answer:
+  Your answer: yarn is a program created by facebook that facilitates building, testing, and debugging React websites.
 
-  Researched answer:
+  Researched answer: yarn is a package manager for code. Running yarn start creates in the current directory the file yarn.lock and the folder node_modules which contains the file .yarn-integrity.
 
 
 
 7. What is localhost:3000? How do we use localhost:3000 in the development process?
 
-  Your answer:
+  Your answer: localhost:3000 is a port on the user's host computer that can display in progress work on a website in the user's browser.
 
-  Researched answer:
+  Researched answer: localhost:3000 is the default port on the user's computer used by React and rails to demo your project while you're working on it.
 
 
 
 8. What is an iterable in JavaScript? What data types are iterables? How can this information useful to you as a developer?
 
-  Your answer:
+  Your answer: Probably something to do with loops since they iterate.
 
-  Researched answer:
+  Researched answer: An iterable in JavaScript is an object that can be iterated over, i.e. it has a series of elements that can be accessed and interacted with sequentially. Strings and arrays are iterable. This property is valuable as it allows to utilize operations (loops, etc.) and and built-in methods (map, filter, etc.) on the iterables.
 
 
 
 9. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: I have seen this word. Related to React. That's all the context my brain can currently give me. Maybe it's importing / exporting classes and methods between parents and children?
 
-  Researched answer:
+  Researched answer: I was thinking of state hoisting which is accomplished by passing a callback from a container component to a child component. (I don't fully understand what those words mean.)
+  
+  But it's also possible that you were talking about hoisting declarations, which is the process that allows variables to be used before they are declared since JavaScript's default behavior is to "hoist" all declarations up to the top of their scope.
 
 
 
 10. STRETCH: What are closures in JavaScript?
 
-  Your answer:
+  Your answer: If I had to guess I would say semicolons. I do not think that this is a good guess.
 
-  Researched answer:
+  Researched answer: A closure is a combination of a function bundled together with references to its surrounding state. In practice this means that a function that resides within another function can access the variables associated with the outer function.
 
 
 
```

Comment:
Oh no! Looks like you might have missed the looking ahead terms. Let's make sure we catch these next time.

Reply:
Didn't realize the looking ahead notes were to be included the assessment, had to convert them from longhand:

React State: An object managed within a component that component's dynamic data. Methods in the program can update the value of the state object and therefore change what appears in the browser.

React Props: Props (properties) is a key word in React for passing information from one component to another. Props are only passed in one direction, from parent to child. Props cannot be updated, they are read-only.

React Lifestyle Methods: The React lifecycle consists of three phases: Mounting, Updating, and Unmounting. Each has a Render phase and a Commit phase. Methods in the Mounting phase are constructor(), static getDerivedStateFromProps(), render(), and componentDidMount. Methods in the Updating phase are staticGetDerivedStateFromProps(), shouldComponentUpdate(), render(), getSnapshotBeforeUpdate(), and componentDidUpdate(). The method in the Unmounting phase is componentWillUnmount().

API: Stands for application programming interface. It is a set of functions and procedures allowing the creation of applications that access the features or data of an operating system, application, or other service.

e.preventDefault(): This is used to prevent a browser's default behavior response to an event. For example it can be used to keep a form from submitting automatically so the data can be sent asynchronously.

DOM events- DOM events allow JavaScript to register different event handlers on elements in an HTML document. Can be used in combo with functions. Functions do not execute before the event occurs.